### PR TITLE
Fix native image doc inconsistency

### LINKED ISSUE #5111: Inconsistent guides for native image
I see some inconsistencies in guides for native image

https://quarkus.io/guides/building-native-image-guide mentions new way using `<quarkus.package.type>native</quarkus.package.type>` property. 

https://quarkus.io/guides/maven-tooling mentions only maven plugin definition way.

https://quarkus.io/guides/writing-native-applications-tips mentions only maven plugin definition way.

https://quarkus.io/guides/maven-tooling#build-tool-maven says: If you have not used project scaffolding, add the following elements in your pom.xml
But project scaffolding now uses `<quarkus.package.type>native</quarkus.package.type>` property way but the guide uses old way.

I think there should be more details about two ways users can take, what are their advantages and how they should select their way. Maybe few words for guys migrating from previous versions.

I played a bit with some text changes but the result wasn't very good so I scratched it and creating issue instead ;)

### PR DESCRIPTION
**Pull request by gsmet**
Fixes #5111

@geoand @gastaldi  I'm not totally sure of what should be adjusted for Gradle but I would appreciate if one of you could have a look at what I did here and see if we need something similar for the Gradle parts of the documentation (we probably do).

Note: the change in the config file is just about reordering the options so that the most useful ones are at the top of the documentation.

### CODE REVIEW

**Comment by geoand** [comment]
@gsmet sure thing! I'll take a look tomorrow since I don't think I'll have time today.

**Comment by gsmet** [comment]
@gsmet OK cool thanks! If you could review this one at the same time you have a look at the Gradle part, that would be nice!

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 1]
```diff
@@ -456,6 +445,16 @@ Uber-Jar creation by default excludes link:https://docs.oracle.com/javase/tutori
...
 
+[[configuration-reference]]
+== Configuring the Project Output
+
+There are a several configuration options that will define what the output of your project build will be.
+These are provided in `application.properties` the same as any other config property.
+
+The properties are shown below:
+
+include::{generated-dir}/config/quarkus-core-package-config.adoc[opts=optional]
```
Comment (machi1990) [inline docs/src/main/asciidoc/maven-tooling.adoc]: 👍

File: docs/src/main/asciidoc/writing-native-applications-tips.adoc [hunk 2]
```diff
@@ -53,26 +53,24 @@ Now, let's tell the `native-image` executable about this configuration file by t
...
+            <quarkus.package.type>native</quarkus.package.type>
+            <quarkus.native.additional-build-args>-H:ResourceConfigurationFiles=${project.basedir}/resources-config.json</quarkus.native.additional-build-args>
+        </properties>
+    </profile>
+</profiles>
 ----
 
+[TIP]
+====
+Multiple options should be separated by a comma.
```
Comment (machi1990) [inline docs/src/main/asciidoc/writing-native-applications-tips.adoc]:
```suggestion
Multiple options can be separated by a comma.
```

Comment (gastaldi) [reply docs/src/main/asciidoc/writing-native-applications-tips.adoc]: IMHO "should" is correct here, since you must use commas in this case

Comment (geoand) [reply docs/src/main/asciidoc/writing-native-applications-tips.adoc]: I agree with @gastaldi

**Comment by geoand** [comment]
Just taking a super quick look at this, I am pretty sure we'll need to corresponding Gradle changes. @gastaldi my availability will be a little flacky this week, do you think you can this one?

File: docs/src/main/asciidoc/writing-native-applications-tips.adoc (hunk 2 above)
Comment (machi1990) [reply docs/src/main/asciidoc/writing-native-applications-tips.adoc]: Yes. When I made the suggestion I found `should` a bit high toned for a tip hence the suggestion.  But thinking about it, the original sentence was right as @gastaldi said. Let's keep it that way.

**Comment by gastaldi** [comment]
@geoand sure, I'll have a look at that Monday morning

File: docs/src/main/asciidoc/writing-native-applications-tips.adoc (hunk 2 above)
Comment (gastaldi) [reply docs/src/main/asciidoc/writing-native-applications-tips.adoc]: Actually `must` is the right term here, for the same reason

**Comment by gsmet** [comment]
I merged this. @gastaldi I let you open a new one with the Gradle adjustments next week?

**Comment by gastaldi** [comment]
Adding to #5101

**Pull request merged**